dot/network/message.go

```diff
@@ -155,21 +155,21 @@ func (bm *BlockRequestMessage) Decode(in []byte) error {
 	}
 
 	var (
-		startingBlock *variadic.Uint64OrHash
+		startingBlock *variadic.Uint32OrHash
 		endBlockHash  *common.Hash
 		max           *uint32
 	)
 
 	switch from := msg.FromBlock.(type) {
 	case *pb.BlockRequest_Hash:
-		startingBlock, err = variadic.NewUint64OrHash(common.BytesToHash(from.Hash))
+		startingBlock, err = variadic.NewUint32OrHash(common.BytesToHash(from.Hash))
 	case *pb.BlockRequest_Number:
-		// TODO: we are receiving block requests w/ 4-byte From field; this should probably be
-		// 4-bytes as it represents a block number which is uint32 (#1854)
-		if len(from.Number) != 8 {
-			return errors.New("invalid BlockResponseMessage.From; uint64 is not 8 bytes")
+		if len(from.Number) != 4 {
+			return fmt.Errorf("invalid BlockRequestMessage.From; expected 4 bytes, got %d bytes", len(from.Number))
```

Comment:
nit - a bit out of scope, but since we are modifying here, can you add a sentinel error for this?

```go
var ErrBlockRequestFromNumberNotValid = errors.New("block request message from number is not valid")
```

and

```suggestion
			return fmt.Errorf("%w: expected 4 bytes, got %d bytes",ErrBlockRequestFromNumberNotValid, len(from.Number))
```

Reply:
updated